pipelines/main/launch_upload_jobs.yml

```diff
@@ -23,6 +23,10 @@ steps:
           export BUILDKITE_PLUGIN_CRYPTIC_BASE64_SIGNED_JOB_ID_SECRET
           export ALLOW_FAIL="false"
 
+          # allow test-failures for non-release/tags so that nightly & PR builds
+          # can be uploaded to s3 even if tests are not passing
+          # (allowing for usage with juliaup)
+          export ALLOW_TEST_FAILURES=$(([[ "${BUILDKITE_BRANCH?}" == release-* ]] || [[ "${BUILDKITE_TAG:-}" == v* ]]) && echo "false" || echo "true")
```

Comment:
I haven't actually verified myself that this pattern matching is correct, so ideally someone would just do some quick tests in Bash locally before we merge.

Or alternatively, we can have a quick GitHub Actions CI job to check miscellaneous things such as this. E.g. just a quick Bash-based CI job that puts in various values of `BUILDKITE_BRANCH` and `BUILDKITE_TAG`, and tests that the output `ALLOW_TEST_FAILURES` is correct.

The latter is probably easier, because we can do it in CI instead of relying on people to manually test things locally.